### ISSUE TITLE
Document/Clarify Removal of GraphTraversal Functionality

### ISSUE THREAD

**Issue by michaelweinold** [brightway-lca/brightway2-calc#91]
@cmutel, you moved (some?) graph traversal functionality to the `bw_graph_tools` package with https://github.com/brightway-lca/brightway2-calc/commit/273c5209ba22f9ddae2ef7e060bd7de0b9d27a8e. This is not documented in the changelog (or anywhere else except in the commit title). The last changelog entry related to graph traversal [simply mentions](https://github.com/brightway-lca/brightway2-calc/blob/main/CHANGES.md#20dev11-2022-08-31):

> Changed GraphTraversal to AssumedDiagonalGraphTraversal

This means that on the current `main` branch, there is __no__ graph traversal functionality in the `bw2calc` package at all.

However, in the latest `dev` version, there is a [`bw2calc/graph_traversal.py`](https://github.com/brightway-lca/brightway2-calc/blob/2.0.DEV16/bw2calc/graph_traversal.py) file . `bw_graph_tools` is not called anywhere.

Is there some reason to this madness 🙈?

 - [ ] On the current version of Brightway2 (`main`), how should one use the `GraphTraversal` class? Import `bw_graph_tools` directly?
 - [ ] On the current version of Brightway25 (`dev`), how should one use the `GraphTraversal` class? Call it from `bw2calc`? What is the role of `bw_graph_tools` in this case?

**Comment by cmutel** [comment]
> However, in the latest dev version, there is a [bw2calc/graph_traversal.py](https://github.com/brightway-lca/brightway2-calc/blob/2.0.dev12/bw2calc/graph_traversal.py) file

You posted a link to 2.0.dev12 - the latest `dev` release is 2.0.dev16 :)

**Comment by michaelweinold** [comment]
...oh - it seems that GH can't sort tags properly:

<img width="344" alt="Screenshot 2024-03-06 at 08 14 45" src="https://github.com/brightway-lca/brightway2-calc/assets/23102087/f20b5e4e-3c8e-42c5-b249-e988e95c4a01">

But in any case, the file is also present in the `dev16` version.

**Comment by cmutel** [comment]
You're right, sorry about that. The DEV16 release was on [October 22](https://github.com/brightway-lca/brightway2-calc/releases/tag/2.0.DEV16), and the graph traversal was removed on [October 26](https://github.com/brightway-lca/brightway2-calc/commit/273c5209ba22f9ddae2ef7e060bd7de0b9d27a8e). 

Brightway2 is pinned to `bw2calc` < 2.0.

> On the current version of Brightway25 (dev), how should one use the GraphTraversal class? Call it from bw2calc? What is the role of bw_graph_tools in this case?

One should only use `bw_graph_tools`, this is strictly better in every way than the old graph traversal class.

**Comment by michaelweinold** [comment]
Ok, thank you for the clarification. Can I therefore:

 - remove the [bw2calc/graph_traversal.py](https://github.com/brightway-lca/brightway2-calc/blob/2.0.DEV16/bw2calc/graph_traversal.py) file from the latest `dev` version
 - and make a corresponding changelog entry

?

**Comment by cmutel** [comment]
Changelog update in https://github.com/brightway-lca/brightway2-calc/commit/1dd1132d92eefe81b6444c8243785dccbd71fa82. But the `main` branch isn't ready for a release yet (yes, yes, this should have been a feature branch and not merged to `main`...).

So I think for now you will just need to manually use `bw_graph_utils` instead.

**Comment by michaelweinold** [comment]
Fine, we'll keep this open until then. Thank you! 
I will update the documentation of the `bw_graph_tools` package in the coming weeks, as I go through the code in detail.
